packages/react-native/native-js-interface.md

```diff
@@ -4,27 +4,143 @@ This is a working spec of this the interface that the native layer (`BugsnagReac
 
 ## Intro
 
-`BugsnagReactNative` is a class which will be instantiated once for the entire runtime and accessible via JS. It's responsible for managing the instantiation of the native client (either `bugsnag-android` or `bugsnag-cocoa`) and providing forwarding on data and invocations from JS to the native client.
+`BugsnagReactNative` is a class which will be instantiated once for the entire runtime and accessible via JS. It's responsible for managing the communicating with the native client (either `bugsnag-android` or `bugsnag-cocoa`) and providing data and invocations from JS to the native client.
 
 ## Accessing `BugsnagReactNative` from JavaScript
 
 Access is set up in the standard React Native way<sup>[1](https://facebook.github.io/react-native/docs/0.59/native-modules-ios),[2](https://facebook.github.io/react-native/docs/0.59/native-modules-android)</sup> such that `BugsnagReactNative` is available in JavaScript like so:
 
 ```js
 const { NativeModules } = require('react-native')
-const NativeClient = NativeModules.BugsnagReactNative
+const { BugsnagReactNative } = NativeModules
 ```
 
 ## API
 
-### `BugsnagReactNative.leaveBreadcrumb(breadcrumb): void`
+### Methods
+
+#### `BugsnagReactNative.getConfig(): configMap`
+
+Retrieves a data structure representing the configuration which the native client was initialised with. The data structure needn't be immutable since it will be passed over the bridge anyway, but mutations should not have any effect.
+
+An "empty" map will indicate to the JS layer that the native client is not configured, which is a problem.
+
+This method is required to be synchronous so it should be annotated as such. This is because during its synchronous configuration, the JS layer needs to obtain config from the native layer.
+
+###### Android
+
+```java
+@ReactMethod(isBlockingSynchronousMethod = true)
+public ReadableMap getConfig()
+```
+
+###### iOS
+```objc
+(NSDictionary *)getConfig;
+
+// annotation goes in BugsnagReactNative.m
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getConfig);
+```
+
+#### `BugsnagReactNative.updateClientState(key, updates): void`
```

Comment:
Since we boiled down clientState to just two updateable properties (that can go over the RN bridge), would it instead make sense to have to separate methods?

- `updateMetaData(key, val)`
- `updateUser(id, name, email)`

The reason I kept it like this is to future-proof – new updateable properties result in changes to implementation but not interface.

Reply:
Agreed - that makes more sense to me, even if it means future new methods if new state data is added.

Reply:
Would this also include `context`?

Reply:
Ok happy to go separate methods.